GeomIO/Geom_PFIO.F90

```diff
@@ -0,0 +1,92 @@
+#include "MAPL_Generic.h"
+
+module mapl3g_GeomPFIO
+   use mapl_ErrorHandling
+   use ESMF
+   use PFIO
+   use mapl3g_geom_mgr
+   use mapl3g_SharedIO
+   implicit none
+   private
+
+   public :: GeomPFIO
+   type, abstract :: GeomPFIO
+      private
+      integer :: collection_id
+      type(MaplGeom), pointer :: mapl_geom
+   contains
+      procedure(I_stage_data_to_file), deferred :: stage_data_to_file
+      procedure :: initialize
+      procedure :: update_time_on_server
+      procedure :: stage_time_to_file
+      procedure, non_overridable :: get_collection_id
+
+   end type GeomPFIO
+
+   abstract interface
+
+     subroutine I_stage_data_to_file(this, bundle, filename, time_index, rc)
+        use esmf
+        import GeomPFIO
+        class(GeomPFIO), intent(inout) :: this
+        type(ESMF_FieldBundle), intent(in) :: bundle
+        character(len=*), intent(in) :: filename
+        integer, intent(in) :: time_index
+        integer, intent(out), optional :: rc
+     end subroutine I_stage_data_to_file
+
+   end interface
+
+contains
+
+   subroutine update_time_on_server(this, time, rc)
+      class(GeomPFIO), intent(inout) :: this
+      type(ESMF_Time), intent(in) :: time
+      integer, intent(out), optional :: rc
+
+      integer :: status
+      type(StringVariableMap) :: var_map
+      type(Variable) :: time_var
+
+      time_var = create_time_variable(time, _RC)
+      call var_map%insert('time',time_var)
+      call o_Clients%modify_metadata(this%collection_id, var_map=var_map, _RC)
+
+      _RETURN(_SUCCESS)
+
+   end subroutine update_time_on_server
+
+   subroutine stage_time_to_file(this,filename, times, rc)
+      class(GeomPFIO), intent(inout) :: this
+      character(len=*), intent(in) :: filename
+      real, intent(in) :: times
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+      type(ArrayReference) :: ref
+
+      ref = ArrayReference(times)
+      call o_Clients%stage_nondistributed_data(this%collection_id, filename, 'time', ref)
```

Comment:
`o_clients` will need to be accessed differently in MAPL3.   But if you can get this to work now, we can fix it later.

Reply:
@tclune what's the right way to do this in MAPL3? Do you have an example somewhere?

Reply:
No.  :-)

The issue is that in MAPL2 we ended up with a global (module) variable for the clients.  And I would like to avoid that.

At the very least we can move this to an accessor on the MaplFramework singleton that I started as part of the GEOS.F90 PR.   Am doing work on that this week to allow our remote server (and an MIT server).   Which brings me closer to the issue I'm highlighting.

Long term, I may pass the MAPL_OBJECT down through the hierarchy which will necessitate an extra argument in lots of places.    I may be satisfied instead by limiting us to a single global variable (THE_MAPL_OBJECT).